snapcraft.yaml

```diff
@@ -136,6 +138,10 @@ parts:
     #source-branch: feature-snapcraft-build
     # We should specify either the source-branch or the source-tag, but not both
     # source-tag: R2019b
+    override-build: |
+      npm install -g npm@next   # Because the the default version in Ubuntu 18.04 is broken
```

Comment:
I was having the problem as of 2 nights ago. I just tried with the line removed and ran into a new issue. The ros deb that was [just released](http://packages.ros.org/ros/ubuntu/db/) seems to use a version of lbssl1.0-dev that conflicts with what npm uses, such that installing ros removes npm. Do you happen to know where to report that issue to ROS?

Reply:
Fyi, I've reported the issue here: https://github.com/ros/rosdistro/issues/27302

Reply:
I think the `npm install -g npm@next` line works around the issue because it creates a `/usr/local/bin/npm` which is used even though the `npm` package has been removed.

Reply:
Ok, thank you for looking deeply at this issue.

Reply:
Based on [discussion](https://github.com/ros/rosdistro/issues/27302) with the ros folks, my latest commit avoids needing to update npm by using specific versions of nodejs and nodejs-dev. This feels like a cleaner solution to me because it doesn't involve creating an entirely separate npm installation in /usr/local. I recommend a similar change to `linux_optional_compilation_dependencies.sh` to force the npm versions to be what I use in the latest commit so that contributors on a stock bionic system will be able to build successfully. Strictly speaking that would mean that the versions wouldn't need to also be specified in snapcraft.yaml, but it doesn't hurt.